apps/aefate/src/aefa_engine_state.erl

```diff
@@ -344,6 +365,20 @@ spend_gas(X, #es{gas = Gas} = ES) ->
         false -> ES#es{gas = NewGas}
     end.
 
+%% The gas price per cell increases by 1 for each kibiword (each 1024 64-bit word) used.
+-spec spend_gas_for_new_cells(non_neg_integer(), state()) -> state().
+spend_gas_for_new_cells(NewCells, #es{ created_cells = Cells } = ES) when NewCells + Cells =< 1024 ->
+    TotalCells = Cells + NewCells,
+    spend_gas(NewCells, ES#es{ created_cells = TotalCells });
+spend_gas_for_new_cells(1, #es{ created_cells = Cells } = ES) ->
+    TotalCells = Cells + 1,
+    CellCost = 1 + (TotalCells bsr 10),
+    spend_gas(CellCost, ES#es{ created_cells = TotalCells });
+spend_gas_for_new_cells(NewCells, #es{ created_cells = Cells } = ES) ->
+    TotalCells = Cells + NewCells,
+    CellCost = 1 + (TotalCells bsr 10),
+    spend_gas(NewCells * CellCost, ES#es{ created_cells = TotalCells }).
```

Comment:
Why special case for `NewCells = 1` and `NewCells + Cells =< 1024`? As far as I can see the general case does the right thing in both of these cases.

Reply:
just wanted to avoid a multiplication in the most used case.

Reply:
Premature optimizations to the rescue.

Reply:
The computation was much more convoluted in previous version, I can tidy it up in a later PR.